svelte/src/scripts/draw/helper/custom-d3-forces.ts

```diff
@@ -1,166 +1,222 @@
-import type { Force, SimulationNodeDatum } from 'd3';
 
-interface simulationNodeDatumType extends SimulationNodeDatum {
-	width: number;
-	height: number;
-	cx?: number;
-	cy?: number;
-}
+import type {Force} from "d3";
+import type { GraphDataNode, GraphDataEdge } from "$types";
 
 /**
  * Creates a rectangular collison force across all nodes
- *
+ * 
  * Assumes all nodes are rectangles, with a width and height property, rather than circles
  */
-export function rectangleCollideForce(): Force<
-	simulationNodeDatumType,
-	d3.SimulationLinkDatum<simulationNodeDatumType>
-> {
-	let nodes: simulationNodeDatumType[];
-
-	interface point {
-		x: number;
-		y: number;
-	}
-
-	interface squareData {
-		topleft: point;
-		bottomright: point;
-		middle: point;
-		width: number;
-		height: number;
-	}
-
-	function makeSquare(node: simulationNodeDatumType): squareData {
-		const topLeftx = node.x! + node.cx!;
-		const topLefty = node.y! + node.cy!;
-		return {
-			topleft: {
-				// Coordinates of top-left point
-				x: topLeftx,
-				y: topLefty
-			},
-			bottomright: {
-				// Coordinates of bottomright point
-				x: topLeftx + node.width,
-				y: topLefty + node.height
-			},
-			middle: {
-				x: topLeftx + node.width,
-				y: topLefty + node.height
-			},
-			width: node.width,
-			height: node.height
-		};
-	}
-
-	function calcForceStrength(node1: squareData, node2: squareData, alpha: number) {
-		return (
-			((node1.width + node1.height) / (node1.width + node1.height + node2.width + node2.height)) *
-			(1 + alpha) *
-			5
-		); // random magic number
-	}
-
-	// actual force calulation
+export function rectangleCollideForce(): Force<GraphDataNode, undefined>  {
+    let nodes: GraphDataNode[];
+    const snap = false;
+
+    interface point {
+        x: number,
+        y: number,
+    }
+    
+    interface squareData {
+        topleft: point,
+        bottomright: point,
+        middle: point,
+        width: number,
+        height: number,
+    }
+    
+    function makeSquare(node: GraphDataNode): squareData {
+        const topLeftx = node.x! + node.cx!;
+        const topLefty = node.y! + node.cy!;
+        return {
+            topleft: { // Coordinates of top-left point
+                x: topLeftx,
+                y: topLefty,
+            },
+            bottomright: { // Coordinates of bottomright point
+                x: topLeftx + node.width,
+                y: topLefty + node.height,
+            },
+            middle: {
+                x: topLeftx + node.width,
+                y: topLefty + node.height,
+            },
+            width: node.width,
+            height: node.height,
+        };
+    };
+    
+    function calcForceStrength(node1: squareData, node2: squareData, alpha: number) {
```

Comment:
hmm what is alpha here? is it intentionally named alpha because it has the same meaning as alpha in the simulation? if not, might be not a bad idea to change the name (to prevent future confusion)

Reply:
Yes, it is the same alpha as in the simlulation